File: app/src/main/java/com/android/akl/bluetoothscreamer/DisconnectionActivity.java
package com.android.akl.bluetoothscreamer;

import android.annotation.SuppressLint;
import android.graphics.Color;
import android.media.MediaPlayer;
import android.os.Handler;
import android.support.constraint.ConstraintLayout;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.TextView;

import java.util.Objects;

public class DisconnectionActivity extends AppCompatActivity {

    ConstraintLayout mConstraintLayout;
    TextView mTextView;
    Button mButton;
    int temp = 0;

    @SuppressLint("SetTextI18n")
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_disconnection);

        mConstraintLayout = findViewById(R.id.disconnection_activity_layout);
        mTextView = findViewById(R.id.disconnection_activity_tv);
        mButton = findViewById(R.id.disconnection_activity_button);

        String deviceName = Objects.requireNonNull(getIntent().getExtras()).getString("device_name");
        mTextView.setText(deviceName + " disconnected");
        final MediaPlayer player = MediaPlayer.create(getApplicationContext(),R.raw.siren_sound);
        player.setVolume(10,10);
        player.start();

        Handler handler = new Handler();
        handler.postDelayed(new Runnable() {
            @Override
            public void run() {
                if(temp == 0){
                    mConstraintLayout.setBackgroundColor(Color.RED);
                    temp = 1;
                }else{
                    mConstraintLayout.setBackgroundColor(Color.WHITE);
                    temp = 0;
                }
            }
        },1000);

        mButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                player.stop();
                finish();
            }
        });
    }
}

File: app/src/main/java/com/android/akl/bluetoothscreamer/DisconnectionActivity.kt
package com.android.akl.bluetoothscreamer

import android.annotation.SuppressLint
import android.app.KeyguardManager
import android.content.Context
import android.graphics.Color
import android.media.AudioManager
import android.media.MediaPlayer
import android.media.RingtoneManager
import android.os.Build
import android.os.Bundle
import android.os.Handler
import android.view.WindowManager
import android.widget.Button
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity
import androidx.constraintlayout.widget.ConstraintLayout


class DisconnectionActivity : AppCompatActivity() {
    private lateinit var mConstraintLayout: ConstraintLayout
    var mTextView: TextView? = null
    var mButton: Button? = null
    var temp = 0
    @SuppressLint("SetTextI18n")
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_disconnection)

        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O_MR1) {
            setShowWhenLocked(true)
            setTurnScreenOn(true)
            val keyguardManager = getSystemService(Context.KEYGUARD_SERVICE) as KeyguardManager
            keyguardManager.requestDismissKeyguard(this, null)
        } else {
            this.window.addFlags(WindowManager.LayoutParams.FLAG_DISMISS_KEYGUARD or
                    WindowManager.LayoutParams.FLAG_SHOW_WHEN_LOCKED or
                    WindowManager.LayoutParams.FLAG_TURN_SCREEN_ON)
        }

        mConstraintLayout = findViewById(R.id.disconnection_activity_layout)
        mTextView = findViewById(R.id.disconnection_activity_tv)
        mButton = findViewById(R.id.disconnection_activity_button)

        val deviceName = intent.extras?.getString("device_name")
        mTextView?.text = "$deviceName disconnected"

        val player = playSound()

        mButton?.setOnClickListener {
            player.stop()
            finish()
        }
    }

    private fun playSound(): MediaPlayer {
        val audioManager = getSystemService(AUDIO_SERVICE) as AudioManager
        audioManager.setStreamVolume(AudioManager.STREAM_MUSIC, 20, 0)
        val player = MediaPlayer.create(applicationContext, RingtoneManager.getActualDefaultRingtoneUri(applicationContext, RingtoneManager.TYPE_RINGTONE))
        player.start()

        val handler = Handler()
        handler.postDelayed({
            temp = when (temp) {
                0 -> {
                    mConstraintLayout.setBackgroundColor(Color.RED)
                    1
                }
                else -> {
                    mConstraintLayout.setBackgroundColor(Color.WHITE)
                    0
                }
            }
        }, 1000)
        return player
    }
}